app/scripts/lib/createEvmMethodsToNonEvmAccountReqFilterMiddleware.ts

```diff
@@ -0,0 +1,70 @@
+import { isEvmAccountType } from '@metamask/keyring-api';
+import { JsonRpcMiddleware } from 'json-rpc-engine';
+import log from 'loglevel';
+import { RestrictedEthMethods } from '../../../shared/constants/permissions';
+import { UnrestrictedEthSigningMethods } from '../controllers/permissions';
+
+/**
+ * Returns a middleware that filters out requests whose requests are restricted to EVM accounts.
+ *
+ * @returns The middleware function.
+ */
+export default function createEvmMethodsToNonEvmAccountReqFilterMiddleware({
+  messenger,
+}: {
+  messenger: any;
```

Comment:
Might be helpful to add more specific typing here since we are making this a ts file
```typescript
interface Messenger {
  call: (method: string, ...args: any[]) => any;
}
```